draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -778,52 +779,76 @@ responses to the address queries that were issued in parallel.
 A few initial SvcParamKeys are defined here.  These keys are useful for
 HTTPS, and most are applicable to other protocols as well.
 
-## "transport" and "no-default-transport" {#transport-key}
+## "alpn" and "no-default-alpn" {#alpn-key}
 
-The "transport" and "no-default-transport" SvcParamKeys together
-indicate the set of transport protocols supported by this service endpoint.
-Each transport protocol is identified by a protocol-id, which is a sequence
-of 1-255 octets.
+The "alpn" and "no-default-alpn" SvcParamKeys together
+indicate the set of Application Layer Protocol Negotation (ALPN)
+protocol names (per {{ALPN=?RFC7301}}) 
+and associated transport protocols supported by this service endpoint.  
 
-    protocol-id = 1*255(OCTET)
+As with {{AltSvc}}, the ALPN protocol name is used to 
+identify the application protocol and associated suite 
+of protocols used by the SVCB record (the "protocol suite").
+Clients use the set of ALPN protocol
+to filter SVCB records to the set of protocol suites they support,
+and then this informs the underlying transport protocol used (such
+as QUIC-over-UDP or TLS-over-TCP).
 
-The presentation value of "transport" is a comma-separated list of one or
-more `protocol-id`s.  Any commas present in the protocol-id are escaped
+The actual Application Layer Protocol negotiated during 
+the secure handshake MAY differ from the ALPN specified 
+in the SVCB record.
+
+ALPNs are identified by their registered "Identification Sequence"
+(alpn-id), which is a sequence of 1-255 octets.
+
+    alpn-id = 1*255(OCTET)
+
+The presentation value of "alpn" is a comma-separated list of one or
+more `alpn-id`s.  Any commas present in the protocol-id are escaped
 by a backslash:
 
    escaped-octet = %x00-2b / "\," / %x2d-5b / "\\" / %5d-%FF
    escaped-id = 1*255(escaped-octet)
-   transport-value = escaped-id *("," escaped-id)
+   alpn-value = escaped-id *("," escaped-id)
 
-In the wire format for "transport", each protocol-id is prefixed by its
+In the wire format for "alpn", each alpn-id is prefixed by its
 length as a single octet, and these length-value pairs are concatenated
 to form the SvcParamValue.  These pairs MUST exactly fill the SvcParamValue;
 otherwise, the SvcParamValue is malformed.
 
-For "no-default-transport", the presentation and wire format values MUST be
+For "no-default-alpn", the presentation and wire format values MUST be
 empty.
```

Comment:
Separately, I think that `no-default-alpn` is the wrong default.  If a server wants to support HTTP/1.1 in this way, the bits it spends on saying that (I count 9 bytes) isn't that much.  I suspect that we'll want to prioritize space for servers that support other protocols in future, and burdening them with an additional attribute is not great.

Reply:
This structure isn't intended to increase efficiency.  The additional attribute is there to make it hard to disable TLS accidentally.  Otherwise, a zone owner could easily write "alpn=h3", observe that everything works in their testing, and deploy without realizing that they've created an outage for all their non-QUIC-capable clients.